## QA Review: PR #196 — REQUEST CHANGES

### ❌ CI FAILURE — tests/client/fallback.test.ts broken

All 4 Node matrix jobs fail (run [24596237452](https://github.com/vinceblank/claude-tempo/actions/runs/24596237452)):

```
AssertionError: expected undefined to be 'active'
tests/client/fallback.test.ts:143  (players[0].status)
tests/client/fallback.test.ts:232  (ensembles[0].conductorStatus)
```

**Root cause:** `src/client/index.ts` was migrated from reading `ClaudeTempoStatus` to `ClaudeTempoAttachmentState`, but the two Vitest test mocks in `tests/client/fallback.test.ts` were not updated. They still supply the old attribute name:

```typescript
// line 133 (Strategy-3 getPlayers fallback):
ClaudeTempoStatus: ['active'],    // stale — client now reads ClaudeTempoAttachmentState

// line 214 (discoverEnsembles fallback):
ClaudeTempoStatus: ['active'],    // stale — same
```

The client finds no `ClaudeTempoAttachmentState`, returns `undefined` for both fields.

**Fix (2 lines):** replace both `ClaudeTempoStatus: ['active']` with `ClaudeTempoAttachmentState: ['attached']`.

Note: `test/tools.test.ts` (Mocha) was correctly updated with a `statusToPhase` synthesizer — the `tests/client/fallback.test.ts` (Vitest, different directory) was missed.

---

### Everything else is clean

| Item | Verdict |
|---|---|
| Phase tag mapping (ensemble.ts + cli/commands.ts) | ✅ |
| Maestro filter (positive COORDINATABLE_PHASES allowlist) | ✅ |
| `includeDisconnected`/`includeStale` semantics | ✅ |
| Undefined phase handling (who-am-i / broadcast / client) | ✅ |
| `MaestroPlayerInfo.status` TODO comment | ✅ |
| Scope (stages.ts / TUI / WIRE-PROTOCOL.md untouched) | ✅ |
| Test skips (3 new `it.skip`, all `TODO(#178)`) | ✅ |
| ClaudeTempoStatus residuals (4 intentional) | ✅ |
| Local mocha (707/49/0) | ✅ |

**Verdict: REJECT** — one 2-line fix in `tests/client/fallback.test.ts`, then re-run CI. All other criteria met.

_(Self-review; GitHub blocks self-request-changes.)_

--- a/tests/client/fallback.test.ts
+++ b/tests/client/fallback.test.ts
@@ -130,7 +130,7 @@ describe('TempoClient.getPlayers — fallback precedence', () => {
             ClaudeTempoEnsemble: ['demo'],
             ClaudeTempoPlayerId: ['dave'],
             ClaudeTempoHostname: ['host-1'],
-            ClaudeTempoStatus: ['active'],
+            ClaudeTempoAttachmentState: ['attached'],
           },
         },
       ],
@@ -140,7 +140,9 @@ describe('TempoClient.getPlayers — fallback precedence', () => {
     const players = await client.getPlayers('demo');
     expect(players).toHaveLength(1);
     expect(players[0].playerId).toBe('dave');
-    expect(players[0].status).toBe('active');
+    // Post-#176: `MaestroPlayerInfo.status` carries the attachment phase value
+    // read from `ClaudeTempoAttachmentState` (mock returns 'attached' above).
+    expect(players[0].status).toBe('attached');
   });
 
   it('returns empty array when all three tiers fail', async () => {
@@ -211,7 +213,7 @@ describe('TempoClient.discoverEnsembles — fallback precedence', () => {
           searchAttributes: {
             ClaudeTempoEnsemble: ['mine'],
             ClaudeTempoIsConductor: [true],
-            ClaudeTempoStatus: ['active'],
+            ClaudeTempoAttachmentState: ['attached'],
           },
         },
         {
@@ -229,7 +231,9 @@ describe('TempoClient.discoverEnsembles — fallback precedence', () => {
     expect(ensembles[0].name).toBe('mine');
     expect(ensembles[0].playerCount).toBe(2);
     expect(ensembles[0].hasConductor).toBe(true);
-    expect(ensembles[0].conductorStatus).toBe('active');
+    // Post-#176: `conductorStatus` reads attachment phase from
+    // `ClaudeTempoAttachmentState` (mock returns 'attached' above).
+    expect(ensembles[0].conductorStatus).toBe('attached');
   });
 
   it('returns empty array when both Global Maestro and workflow list fail', async () => {

